### PR TITLE
Test: Synapse router v2 integration (avax)

### PR DESCRIPTION
**Pull request by mikeyrf**
## Description

Fixes # (issue)

## Checklist

- [ ] New Contracts have been tested
- [ ] Lint has been run
- [ ] I have checked my code and corrected any misspellings

## Summary by CodeRabbit

- **Tests**
  - Added integration tests for the Avalanche network in the `SynapseRouterV2AvalancheIntegrationTestFork` contract, enhancing the reliability of the system.
  - Updated the `SynapseRouterV2IntegrationTest` contract, replacing the `assertEq` function with `assertApproxEqAbs` for more precise testing.

- **Refactor**
  - Adjusted the `initiateBridge` function in the `SynapseRouterV2.Integration.Ethereum.t.sol` file to support the Arbitrum network, expanding the system's compatibility.

- **Style**
  - Added a `console.log` statement in the `checkAddressArrays` function for better debugging visibility.

### CODE REVIEW

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol [hunk 1]
```diff
@@ -0,0 +1,981 @@
...
+    function addExpectedTokens() public virtual override {
+        addExpectedToken(NUSD, "nUSD");
+        addExpectedToken(USDC_E, "USDC.e");
+        addExpectedToken(DAI_E, "DAI.e");
+        addExpectedToken(USDT_E, "USDT.e");
+        addExpectedToken(USDC, "CCTP.USDC");
+        addExpectedToken(NETH, "nETH");
+        addExpectedToken(WETH_E, "WETH.e");
+        addExpectedToken(WAVAX, "WAVAX");
+        addExpectedToken(GMX, "GMX");
```
Comment (ChiTimesChi) [inline test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:70]: Native USDT is missing here - my understanding is that we have all the major tokens to test here?

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol [hunk 2]
```diff
@@ -0,0 +1,981 @@
...
+    }
+
+    function addExpectedBridgeTokens() public virtual override {
+        // add synapse bridge module bridge tokens
+        address[] memory originTokensBridgeStables = new address[](5);
+        originTokensBridgeStables[0] = NUSD;
+        originTokensBridgeStables[1] = DAI_E;
+        originTokensBridgeStables[2] = USDC_E;
+        originTokensBridgeStables[3] = USDT_E;
+        originTokensBridgeStables[4] = USDC;
```
Comment (ChiTimesChi) [inline test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:91]: Swap into nUSD & bridge is also supported for native USDT

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol [hunk 3]
```diff
@@ -0,0 +1,981 @@
...
+        originTokensCCTP[2] = USDC_E;
+        originTokensCCTP[3] = DAI_E;
+        originTokensCCTP[4] = USDT_E;
+
+        address[] memory destinationTokensCCTP = new address[](5);
+        destinationTokensCCTP[0] = USDC;
+        destinationTokensCCTP[1] = NUSD;
+        destinationTokensCCTP[2] = USDC_E;
+        destinationTokensCCTP[3] = DAI_E;
+        destinationTokensCCTP[4] = USDT_E;
```
Comment (ChiTimesChi) [inline test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:161]: Native USDT is supported as both origin and destination asset with SynapseCCTP

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol [hunk 4]
```diff
@@ -0,0 +1,981 @@
...
+
+        SwapQuery memory query = router.getOriginAmountOut(tokenIn, tokenSymbol, amountIn);
+        assertEq(query.routerAdapter, address(router));
+        assertEq(query.tokenOut, USDC);
+        assertEq(query.minAmountOut, amountOut);
+        assertEq(query.deadline, type(uint256).max);
+        assertEq(query.rawParams, getSwapParams(pool, indexFrom, indexTo));
+    }
+
+    /// @dev not enough liquidity to test CCTP max bridged amount out
```
Comment (ChiTimesChi) [inline test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:302]: Add a test here for `router.getOriginAmountOut(USDC, "CCTP.USDC", 10**12+1)`? Should return empty struct.

**Review by ChiTimesChi — COMMENTED** [review]
Suggesting a few tests to add, otherwise LGTM

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol (hunk 1 above)
Comment (mikeyrf) [reply test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:70]: In terms of stables and ETH, yes that's the thought but let me know if you want me to add more besides USDT native

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol (hunk 4 above)
Comment (mikeyrf) [reply test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:302]: Ah true. Good call ty

File: test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol (hunk 1 above)
Comment (ChiTimesChi) [reply test/router/v2/integration/SynapseRouterV2.Integration.Avalanche.t.sol:70]: I think that this list should cover all the important cases:
- "supported" stablecoins (ones swappable to nUSD and/or USDC)
- nETH, WETH
- Wrapped Gas token (if not ETH)
- Any edge case chain-specific tokens (like GMX on Avalanche, but on on Arbitrum)

**Review by ChiTimesChi — APPROVED** [review]
LGTM!

**Pull request merged**